tools/internal_ci/helper_scripts/prepare_build_macos_rc

```diff
@@ -27,62 +34,83 @@ if [ -n "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ]; then
-md5sum -c /tmp/python_installer_checksum.md5
-time sudo installer -pkg ./python-3.7.0-macosx10.9.pkg -target /
-time sudo installer -pkg ./python-3.8.0-macosx10.9.pkg -target /
-
-# set xcode version for Obj-C tests
-sudo xcode-select -switch /Applications/Xcode_9.2.app/Contents/Developer/
-
-# Disable some unwanted dotnet options
-export NUGET_XMLDOC_MODE=skip
-export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=true
-export DOTNET_CLI_TELEMETRY_OPTOUT=true
+if [ "${PREPARE_BUILD_INSTALL_DEPS_RUBY}" == "true" ]
+then
+  set +ex  # rvm script is very verbose and exits with errorcode
+  # Advice from https://github.com/Homebrew/homebrew-cask/issues/8629#issuecomment-68641176
+  brew update && brew upgrade brew-cask && brew cleanup && brew cask cleanup
+  rvm --debug requirements ruby-2.5.0
+  source $HOME/.rvm/scripts/rvm
+  set -e  # rvm commands are very verbose
+  time rvm install 2.5.0
+  rvm use 2.5.0 --default
+  time gem install bundler -v 1.17.3 --no-ri --no-doc
+  time gem install cocoapods --version 1.3.1 --no-ri --no-doc
+  time gem install rake-compiler --no-ri --no-doc
+  rvm osx-ssl-certs status all
+  rvm osx-ssl-certs update all
+  set -ex
+fi
 
-# TODO(jtattermusch): better debugging of clock skew, remove once not needed
-date
+if [ "${PREPARE_BUILD_INSTALL_DEPS_OBJC}" == "true" ]
+then
+  # cocoapods
+  export LANG=en_US.UTF-8
+  # pre-fetch cocoapods master repo's most recent commit only
+  mkdir -p ~/.cocoapods/repos
+  time git clone --depth 1 https://github.com/CocoaPods/Specs.git ~/.cocoapods/repos/master
 
-time git submodule update --init
+  # set xcode version for Obj-C tests
+  sudo xcode-select -switch /Applications/Xcode_9.2.app/Contents/Developer/
+
+  # Needed for ios-binary-size
+  time pip install --user pyyaml pyjwt cryptography requests
+
+  # Store intermediate build files of ObjC tests into /tmpfs
+  # TODO(jtattermusch): add explanation why this is needed
+  mkdir /tmpfs/Build-ios-binary-size
+  ln -s /tmpfs/Build-ios-binary-size src/objective-c/examples/Sample/Build
+  mkdir -p /tmpfs/DerivedData
+  rm -rf ~/Library/Developer/Xcode/DerivedData
+  mkdir -p ~/Library/Developer/Xcode
+  ln -s /tmpfs/DerivedData ~/Library/Developer/Xcode/DerivedData
+fi
+
+if [ "${PREPARE_BUILD_INSTALL_DEPS_PYTHON}" == "true" ]
+then
+  # python
+  time pip install --user virtualenv
+  time pip install --user --upgrade Mako six tox setuptools twisted pyyaml pyjwt cryptography requests
+  export PYTHONPATH=/Library/Python/3.4/site-packages
 
-# Store intermediate build files of ObjC tests into /tmpfs
-mkdir /tmpfs/Build-ios-binary-size
-ln -s /tmpfs/Build-ios-binary-size src/objective-c/examples/Sample/Build
-mkdir -p /tmpfs/DerivedData
-rm -rf ~/Library/Developer/Xcode/DerivedData
-mkdir -p ~/Library/Developer/Xcode
-ln -s /tmpfs/DerivedData ~/Library/Developer/Xcode/DerivedData
+  # make sure md5sum is available
+  brew install coreutils
```

Comment:
This line seems make the presubmit test unhappy. https://source.cloud.google.com/results/invocations/24a6a26e-8bb6-47e8-b138-d4af394b04a2

Reply:
I'll try to fix it.